lib/rules/func-style.js

```diff
@@ -86,8 +115,14 @@ module.exports = {
             nodesToCheck["ArrowFunctionExpression:exit"] = function(node) {
                 const hasThisExpr = stack.pop();
 
-                if (enforceDeclarations && !hasThisExpr && node.parent.type === "VariableDeclarator") {
-                    context.report({ node: node.parent, messageId: "declaration" });
+                if (!hasThisExpr && node.parent.type === "VariableDeclarator") {
+                    if (enforceDeclarations && typeof exportFunctionStyle === "undefined") {
+                        context.report({ node: node.parent, messageId: "declaration" });
+                    }
```

Comment:
Similar to the previous, this would be a false negative:

```js
/*eslint func-style: ["error", "declaration", { "overrides": { "namedExports": "expression" } }]*/

var foo = () => {}; // false negative

```